### PR TITLE
Add methods and events for email verification and password reset

### PR DESCRIPTION
**Pull request by blairlunceford**
## Description

- Add `email_verification.created` and `password_reset.created` events
- Add `GET /user_management/email_verification/:id` endpoint
- Add `GET /user_management/password_reset/:id` and `POST /user_management/password_reset` endpoints
- Deprecate current send password reset endpoint

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[x] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

https://github.com/workos/workos/pull/27414

### CODE REVIEW

File: src/user-management/fixtures/password_reset.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "object": "password_reset",
+    "id": "password_reset_01H5JQDV7R7ATEYZDEG0W5PRYS",
+    "user_id": "user_01H5JQDV7R7ATEYZDEG0W5PRYS",
+    "email": "dane@workos.com",
+    "password_reset_token": "Z1uX3RbwcIl5fIGJJJCXXisdI",
+    "password_reset_url": "https://your-app.com.com/reset-password?invitation_token=Z1uX3RbwcIl5fIGJJJCXXisdI",
```
Comment (mattgd) [inline src/user-management/fixtures/password_reset.json:7]: Should this be the following?
```suggestion
    "password_reset_url": "https://your-app.com/reset-password?token=Z1uX3RbwcIl5fIGJJJCXXisdI",
```

File: src/user-management/fixtures/list-invitations.json [hunk 2]
```diff
@@ -12,7 +12,7 @@
       "organization_id": "org_01H5JQDV7R7ATEYZDEG0W5PRYS",
       "inviter_user_id": null,
       "token": "Z1uX3RbwcIl5fIGJJJCXXisdI",
-      "accept_invitation_url": "https://myauthkit.com/invite?invitation_token=Z1uX3RbwcIl5fIGJJJCXXisdI",
+      "accept_invitation_url": "https://your-app.com.com/invite?invitation_token=Z1uX3RbwcIl5fIGJJJCXXisdI",
```
Comment (mattgd) [inline src/user-management/fixtures/list-invitations.json:15]:
```suggestion
      "accept_invitation_url": "https://your-app.com/invite?invitation_token=Z1uX3RbwcIl5fIGJJJCXXisdI",
```

File: src/user-management/fixtures/invitation.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
   "organization_id": "org_01H5JQDV7R7ATEYZDEG0W5PRYS",
   "inviter_user_id": null,
   "token": "Z1uX3RbwcIl5fIGJJJCXXisdI",
-  "accept_invitation_url": "https://myauthkit.com/invite?invitation_token=Z1uX3RbwcIl5fIGJJJCXXisdI",
+  "accept_invitation_url": "https://your-app.com.com/invite?invitation_token=Z1uX3RbwcIl5fIGJJJCXXisdI",
```
Comment (mattgd) [inline src/user-management/fixtures/invitation.json:12]:
```suggestion
  "accept_invitation_url": "https://your-app.com/invite?invitation_token=Z1uX3RbwcIl5fIGJJJCXXisdI",
```

File: src/user-management/user-management.spec.ts [hunk 4]
```diff
@@ -426,6 +450,53 @@ describe('UserManagement', () => {
...
+      expect(fetchURL()).toContain(
+        `/user_management/password_reset/${passwordResetId}`,
+      );
+      expect(passwordReset).toMatchObject({
+        id: 'password_reset_01H5JQDV7R7ATEYZDEG0W5PRYS',
+        userId: 'user_01H5JQDV7R7ATEYZDEG0W5PRYS',
+        email: 'dane@workos.com',
+        passwordResetToken: 'Z1uX3RbwcIl5fIGJJJCXXisdI',
+        passwordResetUrl:
+          'https://your-app.com.com/reset-password?invitation_token=Z1uX3RbwcIl5fIGJJJCXXisdI',
```
Comment (mattgd) [inline src/user-management/user-management.spec.ts:468]: If my comments above are valid:
```suggestion
          'https://your-app.com/reset-password?token=Z1uX3RbwcIl5fIGJJJCXXisdI',
```

File: src/user-management/user-management.spec.ts [hunk 5]
```diff
@@ -426,6 +450,53 @@ describe('UserManagement', () => {
...
+      expect(fetchBody()).toEqual({
+        email: 'dane@workos.com',
+      });
+      expect(response).toMatchObject({
+        id: 'password_reset_01H5JQDV7R7ATEYZDEG0W5PRYS',
+        userId: 'user_01H5JQDV7R7ATEYZDEG0W5PRYS',
+        email: 'dane@workos.com',
+        passwordResetToken: 'Z1uX3RbwcIl5fIGJJJCXXisdI',
+        passwordResetUrl:
+          'https://your-app.com.com/reset-password?invitation_token=Z1uX3RbwcIl5fIGJJJCXXisdI',
```
Comment (mattgd) [inline src/user-management/user-management.spec.ts:493]: Ditto, if above valid:
```suggestion
          'https://your-app.com/reset-password?token=Z1uX3RbwcIl5fIGJJJCXXisdI',
```

File: src/user-management/serializers/create-password-reset-options.serializer .ts [hunk 6]
```diff
@@ -0,0 +1,10 @@
+import {
```
Comment (mattgd) [inline src/user-management/serializers/create-password-reset-options.serializer .ts:1]: Just a visual bug or is there a space in this file name before `.ts`?
image

**Review by mattgd — COMMENTED** [review]
Looks good overall, just a few minor things.

**Comment by blairlunceford** [comment]
@mattgd thanks for catching my copy pasta errors 🤦‍♀️

**Review by mattgd — APPROVED** [review]

**Pull request merged**